### ISSUE TITLE
A question about the labels

### ISSUE THREAD

**Issue by luna9722** [carpenterlab/unet4nuclei#22]
Hello,
I have a question about the boundary labels, I tried the prprocessing file and tried np.unique for one of the labels and got [0 255] A i not supposed to get 3 elements one for the background, one for the cell and one for the boundary ? to encode them later for the multi class classification for unet 3 classes ?? 
I would be grateful if can someone answer my question.

**Comment by jccaicedo** [comment]
You are correct, boundary labels are supposed to be 3 values. Can you visualize the resulting image and post it here? Also, can you give a reference to the specific notebook and block of code where you think the error is happening?

Thank you!

**Comment by luna9722** [comment]
It is not about an error that I found, I am just trying to understand the annotations and your code.
I took an image from the masks and tried this code from the preprocessing file in your repo.

`annot=skimage.io.imread('/content/drive/MyDrive/BBBC039/masks/IXMtest_A02_s1_w1051DAA7C-7042-435F-99F0-1E847D9B42CB.png')
print(annot.shape)
total_objects=0
if len(annot.shape) == 3:
  annot = annot[:,:,0]
    

annot = skimage.morphology.label(annot)
total_objects += len(np.unique(annot)) - 1
    
annot = skimage.morphology.remove_small_objects(annot, min_size=25)
    
boundaries = skimage.segmentation.find_boundaries(annot)

for k in range(2, 2, 2):
  boundaries = skimage.morphology.binary_dilation(boundaries)
        
label_binary = np.zeros((annot.shape + (3,)))
    

label_binary[(annot == 0) & (boundaries == 0), 0] = 1
label_binary[(annot != 0) & (boundaries == 0), 1] = 1
label_binary[boundaries == 1, 2] = 1
    

skimage.io.imsave('/content/imgg0.png', label_binary)

    
print("Total objects: ",total_objects)
imm=skimage.io.imread('/content/imgg0.png')
print(imm.shape)
print(np.unique(imm))
plt.imshow(imm)
`

ANd this is what i got
![image](https://user-images.githubusercontent.com/48836506/116264655-60f96700-a772-11eb-8f0b-76d60c91119c.png)
and I found it weird to get [0 255] knowing that we have 3 classes. Can you explain it to me ? Thank you in advance.
